pkg/gardenlet/controller/shoot/shoot_control.go

```diff
@@ -384,6 +401,25 @@ func (c *Controller) isSeedReadyForMigration(seed *gardencorev1beta1.Seed) error
 	return health.CheckSeedForMigration(seed, c.identity)
 }
 
+func (c *Controller) countShootBastions(ctx context.Context, shoot *gardencorev1beta1.Shoot, gardenClient kubernetes.Interface) (int, error) {
+	// list all bastions that reference this shoot
+	bastionList := operationsv1alpha1.BastionList{}
+	listOptions := client.ListOptions{Namespace: shoot.Namespace}
```

Comment:
To prevent overloading client/server, you can add `client.Limit(1)` to the list options. You are only interested whether there is at least one bastion, you are not interested in the concrete amount.

Reply:
This comment was not resolved